promql/parser/prettier.go

```diff
@@ -0,0 +1,557 @@
+				if itemp >= keywordIndex && itemp <= closingParenIndex {
+					continue
+				}
+				tempItems = append(tempItems, items[itemp])
+			}
+			items = tempItems
+
+		case IDENTIFIER, METRIC_IDENTIFIER:
+			// Case-1: {__name__="metric_name"} => metric_name
+			// Case-2: {__name__="metric_name", job="prettier"} => metric_name{job="prettier"}
+			itr := skipComments(items, i)
+			if i < 1 || item.Val != "__name__" || items[itr+2].Typ != STRING {
+				continue
+			}
+			var (
+				leftBraceIndex  = itemNotFound
+				rightBraceIndex = itemNotFound
+				labelValItem    = items[itr+2]
+				metricName      = labelValItem.Val[1 : len(labelValItem.Val)-1] // Trim inverted-commas.
+				tmp             []Item
+				skipBraces      bool // For handling metric_name{} -> metric_name.
+			)
+			if isMetricNameNotAtomic(metricName) {
+				continue
+			}
+			for backScanIndex := itr; backScanIndex >= 0; backScanIndex-- {
+				if items[backScanIndex].Typ == LEFT_BRACE {
+					leftBraceIndex = backScanIndex
+					break
+				}
+			}
+			for forwardScanIndex := itr; forwardScanIndex < len(items); forwardScanIndex++ {
+				if items[forwardScanIndex].Typ == RIGHT_BRACE {
+					rightBraceIndex = forwardScanIndex
+					break
+				}
+			}
+			if leftBraceIndex == itemNotFound || rightBraceIndex == itemNotFound {
+				continue
+			}
+			itr = skipComments(items, itr)
+			if items[itr+3].Typ == COMMA {
+				skipBraces = rightBraceIndex-5 == skipComments(items, leftBraceIndex)
+			} else {
+				skipBraces = rightBraceIndex-4 == skipComments(items, leftBraceIndex)
+			}
+			identifierItem := Item{Typ: IDENTIFIER, Val: metricName, Pos: 0}
+			// The code below re-orders the lex items in the items array. The re-ordering of
+			// lex items must occur only when in a key-value pair, the key is `__name__`
+			// and the corresponding value is atomic.
+			//
+			// The leftBraceIndex and rightBraceIndex that form a range. Since the position of items beyond
+			// this range do not change, they are simply copied. However, the items within the range require
+			// re-ordering. So, they are added to the lex item slice after appending the identifier (metric_name)
+			// to the lex item slice.
+			//
+			// Also, we need to avoid `metric_name{}` case while re-ordering a `{__name__="metric_name"}`.
+			// Hence, if the indexes of left and right braces are next to each other, we skip
+			// appending them (braces) to the items slice.
+			for j := range items {
+				if j <= rightBraceIndex && j >= leftBraceIndex {
+					// Before printing the left_brace, we print the metric_name. After this, we check for metric_name{}
+					// situation. If __name__ is the only label_matcher, we skip printing '{' and '}'.
+					if j == leftBraceIndex {
+						tmp = append(tmp, identifierItem)
+						if !skipBraces {
+							tmp = append(tmp, items[j])
+						}
+						continue
+					} else if items[itr+3].Typ == COMMA && j == itr+3 || j >= itr && j < itr+3 {
```

Comment:
1. `else` is not require, you can just have the `if`.
2. Please add relevant `( )` brackets else its hard to read. Also, I think you want to check `j == itr+3` before `items[itr+3].Typ == COMMA` to avoid any panics (even if it won't cause panics, it is uncomfortable to read anyway :)).